From: Felipe Borges <felipeborges@gnome.org>
Date: Thu, 25 Aug 2022 16:08:24 +0200
Subject: sound: Fix sound alert selection

The mapping between sounds and the UI elements was wrong.

Fixes #2026

(cherry picked from commit ff904208cb60a11494f671d93458cdef7bfcf501)
---
 panels/sound/cc-alert-chooser.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/panels/sound/cc-alert-chooser.c b/panels/sound/cc-alert-chooser.c
index 2ccadce..686d8c6 100644
--- a/panels/sound/cc-alert-chooser.c
+++ b/panels/sound/cc-alert-chooser.c
@@ -283,9 +283,9 @@ cc_alert_chooser_init (CcAlertChooser *self)
   alert_name = get_alert_name ();
 
   /* If user has selected an old sound alert, migrate them to click. */
-  if (g_strcmp0 (alert_name, "click") != 0 ||
-      g_strcmp0 (alert_name, "hum") != 0 ||
-      g_strcmp0 (alert_name, "string") != 0 ||
+  if (g_strcmp0 (alert_name, "click") != 0 &&
+      g_strcmp0 (alert_name, "hum") != 0 &&
+      g_strcmp0 (alert_name, "string") != 0 &&
       g_strcmp0 (alert_name, "swing") != 0)
     {
       set_custom_theme (self, "click");
@@ -296,9 +296,9 @@ cc_alert_chooser_init (CcAlertChooser *self)
   if (g_strcmp0 (alert_name, "click") == 0)
     set_button (self, self->click_button, TRUE);
   else if (g_strcmp0 (alert_name, "hum") == 0)
-    set_button (self, self->string_button, TRUE);
+    set_button (self, self->hum_button, TRUE);
   else if (g_strcmp0 (alert_name, "string") == 0)
-    set_button (self, self->swing_button, TRUE);
+    set_button (self, self->string_button, TRUE);
   else if (g_strcmp0 (alert_name, "swing") == 0)
     set_button (self, self->swing_button, TRUE);
   else if (alert_name != NULL)
